pkg/search/controller.go

```diff
@@ -274,58 +416,13 @@ func (c *Controller) updateResourceRegistry(oldObj, newObj interface{}) {
 		klog.V(4).Infof("Ignore ResourceRegistry(%s) update event as spec not changed", newRR.Name)
 		return
 	}
-
-	// TODO: stop resource informers if it is not in the new resource registry
-	resources := c.getResources(newRR.Spec.ResourceSelectors)
-
-	clusters := c.getClusters(newRR.Spec.TargetCluster)
-	clusterSets := make(map[string]struct{}, len(clusters))
-
-	for _, cls := range clusters {
-		v, _ := c.clusterRegistry.LoadOrStore(cls, clusterRegistry{
-			resources:  make(map[schema.GroupVersionResource]registrySet),
-			registries: make(registrySet)})
-		cr := v.(clusterRegistry)
-
-		for _, r := range resources {
-			if cr.resources[r] == nil {
-				cr.resources[r] = make(registrySet)
-			}
-			// add registry record for the resource.
-			cr.resources[r][newRR.GetName()] = struct{}{}
-		}
-		// add registry record for the cluster.
-		cr.registries[newRR.GetName()] = struct{}{}
-		c.clusterRegistry.Store(cls, cr)
-
-		clusterSets[cls] = struct{}{}
-
-		// set backendstore
-		backendstore.AddBackend(cls, newRR.Spec.BackendStore)
-
-		c.queue.Add(cls)
+	clusterSet := sets.New[string]()
+	for _, registry := range []*searchv1alpha1.ResourceRegistry{newRR, oldRR} {
```

Comment:
This controller reconcile the relationship between clusters and registries, what's more the key of the controller's queue is cluster name.
At this point, we don't know cluster changed or not, the only thing we know is if `reflect.DeepEqual(oldRR.Spec, newRR.Spec)` not `true`, the registry must changed.
So it's need to do some reconcile. But, as I mentioned above, the key of the controller queue is cluster name, this leave us only 2 choices, 1 is to queue the cluster name, do reconcile at cluster level, that's what I did; 2 is to introduce another worker, which the key of queue is registry name, then we can reconcile at registry level.
However, if we decice to do using method 2, then we also have to retrieve clusters of old registry spec, and clusters of new registry spec, to find out what changed, then reconcile changed clusters, because finally reconciling multi cluster informer, is what we want.
It seemd not possible to reduce the redunancy, since if cluster-registry relationship not changed, the resources may also be changed, like previously we want to cache `Pod`, but currently we no longer want caching `Pod`, the cluster-registry relation not changed, but the resource did. That's why I use `diff` logic.

Reply:
ok.